compiler/src/dmd/backend/mscoffobj.d

```diff
@@ -649,8 +650,23 @@ void MsCoffObj_term(const(char)[] objfilename)
     BIGOBJ_HEADER header = void;
     IMAGE_FILE_HEADER header_old = void;
 
-    time_t f_timedat = 0;
-    time(&f_timedat);
+    // Use a hash of segment content instead of current timestamp for reproducible builds
+    ubyte[] hashInput;
+    for (segidx_t seg = 1; seg < SegData.length; seg++)
+    {
+        seg_data* pseg = SegData[seg];
+        if (pseg.SDbuf && pseg.SDbuf.length())
+        {
+            // Sample a few bytes from each segment
+            size_t bytesToAdd = pseg.SDbuf.length() >= 16 ? 16 : pseg.SDbuf.length();
+            hashInput.length += bytesToAdd;
+            memcpy(&hashInput[hashInput.length - bytesToAdd], pseg.SDbuf.buf, bytesToAdd);
+        }
+    }
+
+    // Blake3 hash the input and use first 4 bytes for timestamp field
+    ubyte[32] hashValue = blake3(hashInput);
```

Comment:
Blake3 might be a bit overkill if you just need 4 bytes. Perhaps `dmd.root.hash.calcHash` is better for this.